main_test.go

```diff
@@ -124,15 +125,276 @@ func assertEqFile(t *testing.T, a, b string) {
-		if la[i] != lb[i] {
+		// Normalize //line directives to compare only the line numbers, not filenames
+		lineA := normalizeLineDirective(la[i])
+		lineB := normalizeLineDirective(lb[i])
+		
+		if lineA != lineB {
 			t.Errorf("files are different at line(%d)\n%s\n%s\n", i, la[i], lb[i])
 		}
 	}
 }
 
+// normalizeLineDirective replaces the filename in //line directives with "FILE" 
+// so that comparisons focus on line numbers, not temp filenames
+func normalizeLineDirective(line string) string {
+	if strings.HasPrefix(line, "//line ") {
+		parts := strings.Split(line, ":")
+		if len(parts) == 2 {
+			return "//line FILE:" + parts[1]
+		}
+	}
+	return line
+}
+
 func copyFile(t *testing.T, from string) string {
 	f := path.Join(t.TempDir(), time.Now().Format("20060102-150405-")+strconv.Itoa(rand.Int()))
 	fbytes, _ := os.ReadFile(from)
 	os.WriteFile(f, fbytes, 0644)
 	return f
 }
+
+func TestPanicLineNumbers(t *testing.T) {
+	testbin := path.Join(t.TempDir(), "go-instrument-testbin")
+	if err := exec.Command("go", "build", "-cover", "-o", testbin, "main.go").Run(); err != nil {
+		t.Fatalf("Failed to build test binary: %v", err)
+	}
+
+	t.Run("simple panic line numbers preserved", func(t *testing.T) {
+		testPanicScenario(t, testbin, "TestFunc", "testdata/internal/panics.go")
+	})
+
+	t.Run("nested panic line numbers preserved", func(t *testing.T) {
+		testNestedPanicScenario(t, testbin, "Level1", "testdata/internal/panics.go")
+	})
+}
+
+// testPanicScenario tests a simple panic scenario and compares full stack traces
+func testPanicScenario(t *testing.T, testbin, entryFunc, sourceFile string) {
+	tempDir := t.TempDir()
+	
+	// Read the source file content
+	sourceBytes, err := os.ReadFile(sourceFile)
+	if err != nil {
+		t.Fatalf("Failed to read source file: %v", err)
+	}
+	
+	// Create original file and main function
+	origContent := string(sourceBytes) + "\n\nfunc main() {\n\t" + entryFunc + "(context.Background())\n}"
+	origFile := path.Join(tempDir, "test_panic.go")
+	if err := os.WriteFile(origFile, []byte(origContent), 0644); err != nil {
+		t.Fatalf("Failed to write original file: %v", err)
+	}
+
+	// Build and run original version
+	origBinary := path.Join(tempDir, "original_panic")
+	buildCmd := exec.Command("go", "build", "-o", origBinary, "test_panic.go")
+	buildCmd.Dir = tempDir
+	if output, err := buildCmd.CombinedOutput(); err != nil {
+		t.Fatalf("Failed to build original binary: %v\nOutput: %s", err, string(output))
+	}
+
+	origOutput, _ := exec.Command(origBinary).CombinedOutput()
+	
+	// Create instrumented file
+	instrumentedFile := path.Join(tempDir, "test_panic_instrumented.go")
+	if err := os.WriteFile(instrumentedFile, []byte(origContent), 0644); err != nil {
+		t.Fatalf("Failed to write instrumented file: %v", err)
+	}
+
+	// Instrument the file
+	if err := exec.Command(testbin, "-w", "-filename", instrumentedFile).Run(); err != nil {
+		t.Fatalf("Failed to instrument file: %v", err)
```

Comment:
```suggestion
		t.Fatal(err)
```
prefer this minimal syntrax. avoid format strings for t.Error/t.Fatal